### PR TITLE
Fixes for various issues identified by mypy

### PR DESCRIPTION
**Pull request by lkuchenb**

### CODE REVIEW

**Comment by lkuchenb** [comment]
@MoritzHahn1337 @KerstenBreuer @cschu This affected everyone's code I think, would be great if you could skim through the changes and confirm them.

File: tests/integration/utils.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def create_user(
         user_updated = deepcopy(user)
         user_updated.uuid = str(user_obj.uuid)
         user_updated.group_site_id = group_obj.site_id
-        user_updated.group_uuid = str(group_obj.uuid)
+        user_updated.group_uuid_id = str(group_obj.uuid)
```
Comment (cschu) [inline tests/integration/utils.py:181]: why?

File: tests/integration/test_user_management.py [hunk 2]
```diff
@@ -72,22 +72,22 @@ def test_name_change(self, testname:str, executing_user:str, target_user:str, ne
...
-        ("self_group_change", "user_a", "", "group_y", 403),
-        ("self_group_change_by_group_admin", "group_x_admin", "", "group_y", 403),
-        ("other_group_change_by_group_admin", "group_x_admin", "user_a", "group_y", 403),
-        ("other_group_change_by_other_group_admin", "group_x_admin", "user_c", "group_x", 403),
-        ("self_group_change_by_admin", "admin", "", "group_y", 204),
+        ("self_group_change"                         , "user_a"          , None       , "group_y"   , 403),
+        ("self_group_change_by_group_admin"          , "group_x_admin"   , None       , "group_y"   , 403),
+        ("other_group_change_by_group_admin"         , "group_x_admin"   , "user_a"   , "group_y"   , 403),
+        ("other_group_change_by_other_group_admin"   , "group_x_admin"   , "user_c"   , "group_x"   , 403),
+        ("self_group_change_by_admin"                , "admin"           , None       , "group_y"   , 204),
```
Comment (cschu) [inline tests/integration/test_user_management.py:79]: I left that just for you :P

**Review by cschu — APPROVED** [review]
Looks good, except for the `uuid_id` ... ><

File: tests/integration/utils.py (hunk 1 above)
Comment (lkuchenb) [reply tests/integration/utils.py:181]: Technical or philosophical question? 😅

File: tests/integration/test_user_management.py (hunk 2 above)
Comment (lkuchenb) [reply tests/integration/test_user_management.py:79]: I have a script for that, no worries 😁

File: tests/integration/utils.py (hunk 1 above)
Comment (cschu) [reply tests/integration/utils.py:181]: yes.

File: tests/integration/test_user_management.py (hunk 2 above)
Comment (cschu) [reply tests/integration/test_user_management.py:79]: for realz? :O

**Review by MoritzHahn1337 — APPROVED** [review]

**Pull request merged**